kubeprod/pkg/eks/platform.go

```diff
@@ -115,7 +112,7 @@ func (conf *Config) createHostedZone() (*string, error) {
 // Creates a new user policy (or reuses the existing one) in AWS to allow
 // for integration between External DNS and the corresponding hosted zone
 // in Route 53 zone. The user policy is named like "bbkpr-${dnsZone}".
-func (conf *Config) getUserPolicy() (*string, error) {
+func (conf *Config) getUserPolicy(ctx context.Context) (*string, error) {
```

Comment:
Why don't you put the `ctx` variable inside the `conf` structure? This way you save passing it around in functions like `getUserPolicy()`, right?

Reply:
Passing context as the first arg is the strong golang convention.  The convention comes from the fact that there is no way to state in golang that something should _not_ be shared across threads, other than only ever passing it on the stack (as a function argument).

Reply:
See also the intro text from https://golang.org/pkg/context/

Reply:
Ah, I thought this was an AWS Context, not a Golang context. Sorry!